server/middleware/auth.js

```diff
@@ -0,0 +1,52 @@
+var router = require('express').Router(),
+    userModel = require('../db/model/User.js');
+
+module.exports = function(checkType) {
+    type = checkType;
+    router.use(function(req, res, next) {
+        if (req.get('Authorization')) {
+            var authorization = req.get('Authorization');
+            var token = authorization.replace(/Bearer /g, '');
+            userModel.find().byToken(token).exec(function(err, user) {
+                if (!err && user) {
+                    var verify = user.verifyToken(token);
+                    if (verify[0]) {
```

Comment:
Trying to think of another way we could do this without returning a tuple, but since it's not async promise style would be misleading, idk